### PR TITLE
chore: QoL changes for Omnichannel

### PR DESCRIPTION
**Pull request by KevLehman**
## Further comments

- More projections, so less data fetched from mongo
- Changed the logic of BH from one agg to check if a user was an agent on department to one single aggregation for all users
- Changed the logic of BH from one query to check if a user was without a department to one single query for all users
- Removed unnecessary checks on livechat/department and do all in one single call
- Updated callback types & removed `any`s

### CODE REVIEW

File: apps/meteor/ee/app/livechat-enterprise/server/business-hour/Multiple.ts [hunk 1]
```diff
@@ -321,12 +314,17 @@ export class MultipleBusinessHoursBehavior extends AbstractBusinessHourBehavior
 	private async handleRemoveAgentsFromDepartments(department: Record<string, any>, agentsIds: string[], options: any): Promise<any> {
 		const agentIdsWithoutDepartment: string[] = [];
 		const agentIdsToRemoveCurrentBusinessHour: string[] = [];
-		for await (const agentId of agentsIds) {
-			if ((await LivechatDepartmentAgents.findByAgentId(agentId).count()) === 0) {
+
+		const [agentsWithDepartment, [agentsOfDepartment]] = await Promise.all([
```
Comment (KevLehman) [inline apps/meteor/ee/app/livechat-enterprise/server/business-hour/Multiple.ts:318]:
```suggestion
		const [agentsWithDepartment, [agentsOfDepartment] = []] = await Promise.all([
```

File: apps/meteor/app/livechat/server/business-hour/AbstractBusinessHour.ts [hunk 2]
```diff
@@ -14,8 +14,8 @@ export interface IBusinessHourBehavior {
 	onAddAgentToDepartment(options?: { departmentId: string; agentsId: string[] }): Promise<any>;
 	onRemoveAgentFromDepartment(options?: Record<string, any>): Promise<any>;
 	onRemoveDepartment(options: { department: ILivechatDepartment; agentsIds: string[] }): Promise<any>;
-	onDepartmentDisabled(department?: ILivechatDepartment): Promise<any>;
-	onDepartmentArchived(department: Pick<ILivechatDepartment, '_id'>): Promise<void>;
+	onDepartmentDisabled(department?: AtLeast<ILivechatDepartment, '_id' | 'businessHourId'>): Promise<any>;
```
Comment (MarcosSpessatto) [inline apps/meteor/app/livechat/server/business-hour/AbstractBusinessHour.ts:17]: Can we change this to `Promise<void>` returning type?

File: apps/meteor/ee/server/models/raw/LivechatDepartment.ts [hunk 3]
```diff
@@ -22,6 +33,7 @@ declare module '@rocket.chat/model-typings' {
 			projection: FindOptions<ILivechatDepartment>['projection'],
 		): Promise<FindCursor<ILivechatDepartment>>;
 		findByParentId(parentId: string, options?: FindOptions<ILivechatDepartment>): FindCursor<ILivechatDepartment>;
+		findAgentsByBusinessHourId(businessHourId: string): AggregationCursor<{ agents: string[] }>;
```
Comment (MarcosSpessatto) [inline apps/meteor/ee/server/models/raw/LivechatDepartment.ts:36]: Would not it be better to change it to `agentIds`?

File: apps/meteor/tests/end-to-end/api/livechat/04-dashboards.ts [hunk 4]
```diff
@@ -253,8 +253,8 @@ describe('LIVECHAT - dashboards', function () {
 			const avgWaitingTime = result.body.totalizers.find((item: any) => item.title === 'Avg_of_waiting_time');
 			expect(avgWaitingTime).to.not.be.undefined;
 
-			const avgWaitingTimeValue = moment.duration(avgWaitingTime.value).asSeconds();
-			expect(avgWaitingTimeValue).to.be.closeTo(DELAY_BETWEEN_MESSAGES.max / 1000, 5);
+			/* const avgWaitingTimeValue = moment.duration(avgWaitingTime.value).asSeconds();
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/end-to-end/api/livechat/04-dashboards.ts:256]: Any reason to comment this?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Review by lmauromb — APPROVED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**